store/pruning/manager.go

```diff
@@ -0,0 +1,105 @@
+package pruning
+
+import (
+	"cosmossdk.io/log"
+	"cosmossdk.io/store/v2"
+)
+
+// Manager is an abstraction to handle the pruning logic.
+type Manager struct {
+	stateStore      store.VersionedDatabase
+	stateCommitment store.Committer
+
+	logger         log.Logger
+	storeOpts      Option
+	commitmentOpts Option
+
+	chStore      chan struct{}
+	chCommitment chan struct{}
+}
+
+// NewManager creates a new Manager instance.
+func NewManager(
+	stateStore store.VersionedDatabase,
+	stateCommitment store.Committer,
+	logger log.Logger,
+) *Manager {
+	return &Manager{
+		stateStore:      stateStore,
+		stateCommitment: stateCommitment,
+		logger:          logger,
+		storeOpts:       DefaultOptions(),
+		commitmentOpts:  DefaultOptions(),
+	}
+}
+
+// SetStoreOptions sets the store options.
+func (m *Manager) SetStoreOptions(opts Option) {
+	m.storeOpts = opts
+}
+
+// SetCommitmentOptions sets the commitment options.
+func (m *Manager) SetCommitmentOptions(opts Option) {
+	m.commitmentOpts = opts
+}
+
+// Start starts the manager.
+func (m *Manager) Start() {
+	m.chStore = make(chan struct{}, 1)
+	m.chStore <- struct{}{}
+	m.chCommitment = make(chan struct{}, 1)
+	m.chCommitment <- struct{}{}
+}
+
+// Stop stops the manager and waits for all goroutines to finish.
+func (m *Manager) Stop() {
```

Comment:
I highly recommend always using sync.Once to protect such operations to prevent future headaches despite documenting that calling .Close or .Stop more than once could panic.